app/models/character.rb

```diff
@@ -61,9 +61,9 @@ def self.deletable_by?(user, user_id)
 
   def recent_posts
     return @recent unless @recent.nil?
-    reply_ids = replies.group(:post_id).pluck(:post_id)
-    post_ids = posts.select(:id).map(&:id)
-    @recent ||= Post.where(id: (post_ids + reply_ids).uniq).ordered
+    reply_ids = replies.select(:post_id).distinct
+    post_ids = posts.map(&:id)
```

Comment:
This looks to be loading all this character's posts, with all associated model details - is that intentional?

Reply:
...huh, I assumed it wouldn't load everything until it was actually called?

Reply:
Unfortunately `.map` is one of the Relation methods that loads everything and starts applying attribute calls to each model in the list - `.pluck` probably works better here, or `.select(:id)`, like you did elsewhere?

Reply:
...I'm confused on why that map is still there, yup. I think I might have deleted the wrong thing. Good catch, I didn't even see that xD